domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -677,8 +677,19 @@ class ExplorationProgressController @Inject constructor(
             userAnswer.answer,
             userAnswer.writtenTranslationContext
           ).outcome
-        answerOutcome =
-          explorationProgress.stateGraph.computeAnswerOutcomeForResult(topPendingState, outcome)
+        val wasVisited = explorationProgress.stateDeck.wasStateVisitedBefore(outcome.destStateName)
+        explorationProgress.stateDeck.setRevisionIndexIfVisited(outcome.destStateName)
```

Comment:
Why is this needed? If it's for checkpoints, we may be able to remove it (see my later comment). If not, I'm not sure that I see why it's necessary.